plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -222,30 +224,22 @@ public void registerModelMeta(MLRegisterModelMetaInput mlRegisterModelMetaInput,
                     GetRequest getModelGroupRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
                     client.get(getModelGroupRequest, ActionListener.wrap(modelGroup -> {
                         if (modelGroup.isExists()) {
-                            Map<String, Object> source = modelGroup.getSourceAsMap();
-                            int latestVersion = (int) source.get(MLModelGroup.LATEST_VERSION_FIELD);
-                            int newVersion = latestVersion + 1;
-                            source.put(MLModelGroup.LATEST_VERSION_FIELD, newVersion);
-                            source.put(MLModelGroup.LAST_UPDATED_TIME_FIELD, Instant.now().toEpochMilli());
-                            UpdateRequest updateModelGroupRequest = new UpdateRequest();
-                            long seqNo = modelGroup.getSeqNo();
-                            long primaryTerm = modelGroup.getPrimaryTerm();
-                            updateModelGroupRequest
-                                .index(ML_MODEL_GROUP_INDEX)
-                                .id(modelGroupId)
-                                .setIfSeqNo(seqNo)
-                                .setIfPrimaryTerm(primaryTerm)
-                                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                                .doc(source);
-                            client
-                                .update(
-                                    updateModelGroupRequest,
-                                    ActionListener
-                                        .wrap(r -> { uploadMLModelMeta(mlRegisterModelMetaInput, newVersion + "", listener); }, e -> {
-                                            log.error("Failed to update model group", e);
-                                            listener.onFailure(e);
-                                        })
-                                );
+                            Map<String, Object> modelGroupSource = modelGroup.getSourceAsMap();
+                            int updatedVersion = incrementLatestVersion(modelGroupSource);
+                            UpdateRequest updateModelGroupRequest = createUpdateModelGroupRequest(
+                                modelGroupSource,
+                                modelGroupId,
+                                modelGroup.getSeqNo(),
+                                modelGroup.getPrimaryTerm(),
+                                updatedVersion
+                            );
+
+                            client.update(updateModelGroupRequest, ActionListener.wrap(r -> {
+                                uploadMLModelMeta(mlRegisterModelMetaInput, updatedVersion + "", listener);
```

Comment:
regarding : `updatedVersion + ""` is the goal is to convert int to string, may be we can follow standard way like: `Integer.toString(updatedVersion)`?